core/src/window_service.rs

```diff
@@ -140,13 +140,20 @@ where
     let mut i = 0;
     shreds.retain(|shred| (verify_repair(&shred, &repair_infos[i]), i += 1).0);
```

Comment:
I think we should also probably run the `retain` on the indexes in `repair_infos` here as well so that the shreds and `repair_infos` still line up